client_code/persistence.py

```diff
@@ -126,7 +126,7 @@ def __getattr__(self, key):
         if self._delta and key in self._delta:
             return self._delta[key]
 
-        return dict(self._store).get(key, None)
+        return self._store.get(key, None)
```

Comment:
but if a `row` is a store we don't have a `.get()`?

Reply:
shite. That's exactly what it was there for. Must slow down... Must slow down...

Reply:
```suggestion
        try:
            self._store[key]
        except (KeyError, anvil.tables.TableError):
            raise AttributeError(key)
```

then adjust `__getitem__`?

Reply:
yeah, I like that. Much nicer. Why would `__getitem__` need to change?

Reply:
To throw a KeyError

Reply:
Right - just so that attribute access raises an AttributeError but dictlike access raises a KeyError?

Reply:
Yeah. That was my thinking.

Reply:
Gotcha. I couldn't work out what logic needed to change!